src/main/java/org/cardanofoundation/tools/adahandle/resolver/storage/AdaHandleStorage.java

```diff
@@ -56,15 +63,18 @@ public Optional<List<AddressUtxo>> saveAll(List<AddressUtxo> addressUtxoList) {
             adaHandleHistoryService.saveAdaHandleHistoryItems(addressUtxoEntities);
         }
 
-        addressUtxoEntities = utxoRepository.saveAll(new ArrayList<>());
-        return Optional.of(addressUtxoEntities.stream()
-                .map(mapper::toAddressUtxo)
-                .toList());
+        utxoRepository.saveAll(new ArrayList<>());
     }
 
     @Override
-    public int deleteBySlotGreaterThan(Long slot) {
+    public void saveSpent(List<TxInput> txInputs) {
+        super.saveSpent(txInputs);
+    }
+
+    @Override
+    public int deleteUnspentBySlotGreaterThan(Long slot) {
         adaHandleHistoryService.rollbackToSlot(slot);
-        return super.deleteBySlotGreaterThan(slot);
+        return super.deleteUnspentBySlotGreaterThan(slot);
```

Comment:
What is the value of overwriting the same thing from the parent?

Reply:
in this case you need both your custom code and the default code.

Indexer saves both set of relevant utxos following the default yaci-storage workflow and custom data.